[Modules] Format the metadata.json and aligned the Name and Description for all Modules and Updated Readme (#3309)
Azure/ResourceModules: modules/network/network-managers/security-admin-configurations/rule-collections/README.md

> # AVNM Security Admin Configuration Rule Collections `[Microsoft.Network/networkManagers/securityAdminConfigurations/ruleCollections]`

Reviewer: ```suggestion
# Network Manager Security Admin Configuration Rule Collections `[Microsoft.Network/networkManagers/securityAdminConfigurations/ruleCollections]`
```

> This module deploys an Azure Virtual Network Manager (AVNM) Security Admin Configuration Rule Collection.

Reviewer: ```suggestion
This module deploys a Network Manager Security Admin Configuration Rule Collection.
```
